src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -449,7 +449,7 @@ private IReadOnlyList<ObjectPropertyInitializer> GetPropertyInitializers(IReadOn
                 {
                     // For structs all properties become required
                     Constant? defaultParameterValue = null;
-                    if (property.SchemaProperty?.ClientDefaultValue is { } defaultValueObject)
+                    if (property.InputModelProperty?.DefaultValue is object defaultValueObject)
```

Comment:
would we write in this way and the type of this local variable `defaultValueObject` changes to `object` type?
Or it looks like to be.
I think we should still use `{}` which is just asserting this is not null

Reply:
reverted